packages/dynamite/dynamite_runtime/lib/src/client/response.dart

```diff
@@ -122,10 +123,24 @@ final class ResponseConverter<B, H> with Converter<http.StreamedResponse, Future
 
     final headers = _deserialize<H>(rawHeaders, serializer.serializers, serializer.headersType);
 
-    final rawBody = switch (serializer.bodyType) {
-      const FullType(Uint8List) => response.bodyBytes,
-      const FullType(String) => response.body,
-      _ => json.decode(response.body),
+    final contentType = rawHeaders['content-type'];
+    MediaType? mediaType;
+    if (contentType != null) {
+      try {
+        mediaType = MediaType.parse(contentType);
+      } on FormatException catch (error, stackTrace) {
+        log('Could not parse $contentType', error: error, stackTrace: stackTrace);
+      }
+    }
+
+    final rawBody = switch (mediaType) {
+      MediaType(type: 'text') || MediaType(type: 'application', subtype: 'javascript') => response.body,
+      MediaType(type: 'application', subtype: 'json') => json.decode(response.body),
+      MediaType(type: '*', subtype: '*') ||
+      MediaType(type: 'application', subtype: 'octet-stream') ||
+      MediaType(type: 'image') ||
+      _ =>
+        response.bodyBytes,
     };
     final body = _deserialize<B>(rawBody, serializer.serializers, serializer.bodyType);
```

Comment:
Do we even need to deserialize in case the string/bytes body is used? For those cases the bodyType should already be Uint8List or String :thinking:

Reply:
I'm not 100% sure with the ContentString but we should be able to do so

Reply:
All tests still pass when changing this :)